frontend/src/actions/index.tsx

```diff
@@ -17,4 +17,30 @@ export interface DecrementAction extends Action {
   type: CounterTypes.DECREASE_COUNT
 }
 
-export type CounterActions = IncrementAction | DecrementAction;
\ No newline at end of file
+export type CounterActions = IncrementAction | DecrementAction
+
+export const SET_USER = 'SET_USER'
+export const LOG_OUT = 'LOG_OUT'
+
+export interface UserDetailsOnRegister {
+  email: string
+  username: string
+  password: string
+}
+
+export interface UserDetailsOnLogin {
+  email: string
+  password: string
+}
+
+export interface fetchUserAction {
```

Comment:
- I don't see any sense of using `typeof` to define `type` in action object. It's an almost standard thing, so no need to use `typeof` 
 - I think can define a generic for actions, so we could use it like `type FetchUser = Action<AuthActionType.SET_USER, UserDetailsOnRegister>`